spacy/pipeline/coref.py

```diff
@@ -511,15 +509,20 @@ def update(
         set_dropout_rate(self.model, drop)
 
         total_loss = 0
-
-        for eg in examples:
-            preds, backprop = self.model.begin_update([eg.predicted])
-            score_matrix, mention_idx = preds
-
-            loss, d_scores = self.get_loss([eg], score_matrix, mention_idx)
+        docs = [eg.predicted for eg in examples]
+        for doc, eg in zip(docs, examples):
+            # replicates the EntityLinker's behaviour and
+            # copies annotations over https://bit.ly/3iweDcW
+            # takes 'coref_head_clusters' from the reference.
+            for key, sg in eg.reference.spans.items():
+                if key.startswith(self.input_prefix):
+                    aligned_spans = eg.get_aligned_spans_x2y(sg)
+                    doc.spans[key] = [doc[span.start:span.end] for span in aligned_spans]
+            span_scores, backprop = self.model.begin_update([doc])
+            loss, d_scores = self.get_loss([eg], span_scores)
             total_loss += loss
             # TODO check shape here
-            backprop((d_scores, mention_idx))
+            backprop(d_scores)
```

Comment:
Doesn't this need a corresponding change in the backprop function, or was that out of sync?

Reply:
The `mention_idx` was already unused. Its an integer array so doesn't do anything in the backprop. The torch `autograd` actually errors out if you give it an int array so it was just wrong, but unused.